docs/canonicalk8s/snap/reference/disa-stig-audit.md

```diff
@@ -5446,1899 +5557,1769 @@ Restart the `kube-apiserver` service:
+grep -E -qvz '\-\-feature-gates=(.*PodSecurity=false.*|.*PodSecurity=0.*)' '/var/snap/k8s/common/args/kubelet' && echo UNSET
 test $? -eq 0 && echo PASS || echo FAIL
 ```
 
 In the default configuration of the `k8s-snap`, resulting output lines will
-start with `PASS`.
+start with `UNSET`.
+
+The final line of the output will be `PASS`.
 
 
-## [V-242448]
+## Bootstrap class 
 
-**Guideline:** The Kubernetes Kube Proxy kubeconfig must be owned by root
+### [V-242384]
+
+**Guideline:** The Kubernetes Scheduler must have secure binding
 
 **Severity:** Medium
 
 **Class**: Default
 
 **Upstream finding description:**
 
-> The Kubernetes Kube Proxy kubeconfig contain the argument and setting for the
-Control Planes. These settings contain network rules for restricting network
-communication between pods, clusters, and networks. If these files can be
-changed, data traversing between the Kubernetes Control Panel components would
-be compromised. Many of the security settings within the document are
-implemented through this file.
-
+> Limiting the number of attack vectors and implementing authentication and
+> encryption on the endpoints available to external sources is paramount when
+> securing the overall Kubernetes cluster. The Scheduler API service exposes
+> port 10251/TCP by default for health and metrics information use. This port
+> does not encrypt or authenticate connections. If this port is exposed
+> externally, an attacker can use this port to attack the entire Kubernetes
+> cluster. By setting the bind address to localhost (i.e., 127.0.0.1), only
+> those internal services that require health and metrics information can
+> access the Scheduler API.
 
 
 
-### Step 1/2
 
 
 **Comments:**
 
-> Finding stipulates the file should be owned by the root user/group.
+> The command line arguments of the Kubernetes Scheduler in the k8s-snap are
+> defined in the following file:
 >
-> Finding V-242447 defines the associated file permissions requirements.
+>     /var/snap/k8s/common/args/kube-scheduler
 >
+> The necessary argument is already set when following the
+> [DISA STIG hardening deployment guide].
 
 
-### Remediation for Step 1
-
-Ensure all of the following paths have correct ownership by running:
-
-
-    chown -R 0:0 /etc/kubernetes/proxy.conf
-
-
-### Auditing (as root) for Step 1
-
-All files exist and have the correct ownership.
-
-```bash
-stat -c %u:%g '/etc/kubernetes/proxy.conf' | grep -q 0:0 && echo PASS /etc/kubernetes/proxy.conf: 0:0 || echo FAIL /etc/kubernetes/proxy.conf: 0:0
-```
-
-In the default configuration of the `k8s-snap`, resulting output lines will
-start with `PASS`.
-
+Remediation
 
+Pre-deployment:
 
-### Step 2/2
+Use the template presented in the [DISA STIG hardening deployment guide] to
+apply the argument automatically to the kubelet.
 
+Post-deployment:
 
-**Comments:**
+Alternatively, if you have already deployed the k8s-snap, manually
+apply the argument by editing the appropriate file:
```

Comment:
This is a little unnecessary to state edit the appropriate file and then edit this file. We could instead shorten it to :
```
Alternatively, if you have already deployed the k8s-snap, manually
apply the argument by editing `/var/snap/k8s/common/args/kube-scheduler` ....
```